### ISSUE TITLE
FormatException: Missing extension byte

### ISSUE THREAD

**Issue by pierremrtn** [tempo-riz/deepgram_speech_to_text#10]
I'm getting an error sometime while using speech-to-text with stream:

```bash
E/flutter ( 5983): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: FormatException: Missing extension byte (at offset 8)
E/flutter ( 5983): #0      _Utf8Decoder.convertSingle (dart:convert-patch/convert_patch.dart:1760:7)
E/flutter ( 5983): #1      Utf8Decoder.convert (dart:convert/utf.dart:349:37)
E/flutter ( 5983): #2      Utf8Codec.decode (dart:convert/utf.dart:63:20)
E/flutter ( 5983): #3      toUt8 (package:deepgram_speech_to_text/src/utils.dart:25:15)
E/flutter ( 5983): #4      DeepgramSttResult.transcript (package:deepgram_speech_to_text/src/deepgram.dart:263:12)
E/flutter ( 5983): #5      SpeechToText._onTranscriptionResult (package:nutriscan/log/speech_to_text.dart:63:45)
E/flutter ( 5983): #6      _RootZone.runUnaryGuarded (dart:async/zone.dart:1594:10)
E/flutter ( 5983): #7      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:365:11)
E/flutter ( 5983): #8      _DelayedData.perform (dart:async/stream_impl.dart:541:14)
E/flutter ( 5983): #9      _PendingEvents.handleNext (dart:async/stream_impl.dart:646:11)
E/flutter ( 5983): #10     _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:617:7)
E/flutter ( 5983): #11     _microtaskLoop (dart:async/schedule_microtask.dart:40:21)
E/flutter ( 5983): #12     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49:5)
```

I'm still able to receive transcription, but this error happen randomly during the transcription.
I'm using deepgram_speech_to_text: ^2.2.1 and record: 5.0.0

```dart
Stream<List<int>> micStream = await AudioRecorder().startStream(
        const RecordConfig(
          encoder: AudioEncoder.pcm16bits,
          sampleRate: 16000,
          numChannels: 1,
        ),
      );
      final streamParams = {
        'detect_language': false, // not supported by streaming API
        'language': language,
        // must specify encoding and sample_rate according to the audio stream
        'encoding': 'linear16',
        'sample_rate': 16000,
      };

      _transcriber = deepgram.createLiveTranscriber(
        micStream,
        queryParams: streamParams,
      );
      _sub = _transcriber!.stream.listen(_onTranscriptionResult);
      _transcriber!.start();
```

Flutter doctor:
```bash
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.22.2, on macOS 14.4.1 23E224 darwin-arm64, locale
    en-FR)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.90.2)
[✓] Connected device (4 available)
[✓] Network resources
```

**Comment by tempo-riz** [comment]
hey @pierremrtn thanks for the report, I will look into it, on what platform are you testing ?

**Comment by tempo-riz** [comment]
The issue is fixed in version 2.2.2. 

To explain a bit more: I previously added UTF-8 conversion because sometimes the Deepgram API returns weirdly encoded characters. 

But since your error happens now if the conversion fails, it returns the text as it is.
